test/system/220-healthcheck.bats

```diff
@@ -273,4 +273,182 @@ Log[-1].Output   | \"Uh-oh on stdout!\\\nUh-oh on stderr!\\\n\"
+    is "$output" "$expect" "$expect_msg"
+
+    output=$cid
+}
+
+function _check_health_log {
+    local ctrname="$1"
+    local expect_msg="$2"
+    local expect_count="$3"
+
+    run_podman inspect $ctrname --format "{{.State.Health.Log}}"
+    count=$(grep -co "$expect_msg" <<< "$output")
+    assert "$count" -eq $expect_count "Number of matching health log messages"
+}
+
+@test "podman healthcheck --health-max-log-count default value (5)" {
+    local repeat_count=10
+    local msg="healthmsg-$(random_string)"
+    local ctrname="c-h-$(safename)"
+    _create_container_with_health_log_settings $ctrname $msg "{{.Config.HealthMaxLogCount}}" "" "5" "HealthMaxLogCount is the expected default"
+
+    for i in $(seq 1 $repeat_count);
+    do
+        run_podman healthcheck run $ctrname
+        is "$output" "" "unexpected output from podman healthcheck run (pass $i)"
+    done
+
+    _check_health_log $ctrname $msg 5
+
+    run_podman rm -t 0 -f $ctrname
+}
+
+@test "podman healthcheck --health-max-log-count infinite value (0)" {
+    local repeat_count=10
+    local msg="healthmsg-$(random_string)"
+    local ctrname="c-h-$(safename)"
+    _create_container_with_health_log_settings $ctrname $msg "{{.Config.HealthMaxLogCount}}" "--health-max-log-count 0" "0" "HealthMaxLogCount"
+
+    # This is run 11 times to check that the cap is working.
+    for i in $(seq 0 $repeat_count);
+    do
+        run_podman healthcheck run $ctrname
+        is "$output" "" "unexpected output from podman healthcheck run (pass $i)"
+    done
+
+    # The healthcheck is triggered by the podman when the container is started, but its execution depends on systemd.
+    # And since `run_podman healthcheck run` is also run manually, it will result in two runs.
+    _check_health_log $ctrname $msg 12
+
+    run_podman rm -t 0 -f $ctrname
+}
+
+
+@test "podman healthcheck --health-max-log-count 10" {
+    local repeat_count=10
+    local msg="healthmsg-$(random_string)"
+    local ctrname="c-h-$(safename)"
+    _create_container_with_health_log_settings $ctrname $msg "{{.Config.HealthMaxLogCount}}" "--health-max-log-count  $repeat_count" "$repeat_count" "HealthMaxLogCount"
+
+    # This is run 11 times to check that the cap is working.
+    for i in $(seq 0 $repeat_count);
+    do
+        run_podman healthcheck run $ctrname
+        is "$output" "" "unexpected output from podman healthcheck run (pass $i)"
+    done
+
+    _check_health_log $ctrname $msg $repeat_count
+
+    run_podman rm -t 0 -f $ctrname
+}
+
+@test "podman healthcheck --health-max-log-size 10" {
+    local msg="healthmsg-$(random_string)"
+    local ctrname="c-h-$(safename)"
+    _create_container_with_health_log_settings $ctrname $msg "{{.Config.HealthMaxLogSize}}" "--health-max-log-size 10" "10" "HealthMaxLogSize"
+
+    run_podman healthcheck run $ctrname
+    is "$output" "" "output from 'podman healthcheck run'"
+
+    _check_health_log $ctrname "healthmsg-}]$" 1
```

Comment:
Suggestion: `substr=${msg:0:10}` then `_check ... "$substr}]\$"`. With an explanation that `max-log-size` truncates the echo message, because there are just way too many options and way too many magic "10"s all over, and it's confusing...